### PR TITLE
bcftools/annotate ready2go

### LINKED ISSUE #1268: new module: bcftools/annotate
I think it would be good to have a module for [bcftools/annotate](https://bioconda.github.io/recipes/bcftools/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by projectoriented**
## PR checklist

Closes #1268  

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/bcftools/annotate/main.nf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    def matcher = input =~ /vcf/
+    def output_suffix = matcher ? "vcf.gz" : "bcf"
+    def output_type_compressed = matcher ? "z" : "b"
+    """
+    bcftools \\
+        annotate \\
+        $args \\
+        --output ${prefix}_annotated.${output_suffix} \\
+        --output-type $output_type_compressed \\
```
Comment (sateeshperi) [inline modules/bcftools/annotate/main.nf:33]: this looks like an optional input....better to let user provide in $args

Comment (projectoriented) [reply modules/bcftools/annotate/main.nf:33]: Hello 👋 , I understand. However, here explains what I did on default 😆 : https://nf-co.re/developers/adding_modules and cntrl find: "Where applicable, the usage and generation of compressed files SHOULD be enforced as input and output, respectively:"

Comment (sateeshperi) [reply modules/bcftools/annotate/main.nf:33]: makes sense. ur good then

**Review by sateeshperi — APPROVED** [review]
lgtm. Thank you for this module!

File: modules/bcftools/annotate/main.nf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def matcher = input =~ /vcf/
+    def output_suffix = matcher ? "vcf.gz" : "bcf"
```
Comment (mahesh-panchal) [inline modules/bcftools/annotate/main.nf:26]:
```suggestion
    def output_suffix = input ==~ /\S+\.*vcf\.\S*/ ? "vcf.gz" : "bcf"
```
I don't see `matcher` used anywhere else so just use it in the ternary operator.
ps. difference between match and find operator: https://groovy-lang.org/operators.html#_match_operator.
Also a match is from the beginning to the end of a string, so provide a full pattern.

File: modules/bcftools/annotate/main.nf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        annotate \\
+        $args \\
+        --output ${prefix}_annotated.${output_suffix} \\
+        --output-type $output_type_compressed \\
+        --threads $task.cpus \\
+        $input
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        bcftools: \$(bcftools --version 2>&1 | head -n1 | sed 's/^.*bcftools //; s/ .*\$//')
```
Comment (mahesh-panchal) [inline modules/bcftools/annotate/main.nf:39]:
```suggestion
        bcftools: \$( bcftools --version |& sed '1!d; s/^.*bcftools //' )
```

**Review by mahesh-panchal — COMMENTED** [review]
Couple of small things.

**Pull request merged**